lib/validator/v3/validator.go

```diff
@@ -1412,9 +1412,13 @@ func validateBGPConfigurationSpec(structLevel validator.StructLevel) {
 		}
 	}
 
+	if (spec.PrefixAdvertisements == nil || len(spec.PrefixAdvertisements) == 0) && (communities != nil || len(communities) != 0) {
```

Comment:
The `== nil` conditions are superfluous, because https://play.golang.org/p/eswiL5pZVMc